### PR TITLE
DOCAPI-7745: optimize_throw_if_noop docs.

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Documentation

### CODE REVIEW

File: docs/en/interfaces/formats.md [hunk 1]
```diff
@@ -934,7 +934,7 @@ You can select data from a ClickHouse table and save them into some file in the
 clickhouse-client --query="SELECT * FROM {some_table} FORMAT Parquet" > {some_file.pq}
 ```
 
-To exchange data with the Hadoop, you can use [`HDFS` table engine](../../operations/table_engines/hdfs.md).
+To exchange data with the Hadoop, you can use [`HDFS` table engine](../operations/table_engines/hdfs.md).
```
Comment (blinkov) [inline docs/en/interfaces/formats.md]: why it's in backticks?

File: docs/en/operations/settings/settings.md [hunk 2]
```diff
@@ -860,4 +860,17 @@ Possible values:
 
 Default value: 0.
 
+## optimize_throw_if_noop {#setting-optimize_throw_if_noop}
+
+Enables or disables throwing an exception if the [OPTIMIZE](../../query_language/misc.md#misc_operations-optimize) query have not performed a merge.
+
+If the `OPTIMIZE` query doesn't perform a merge of data parts for any reason (for example, nothing to merge or lack of resources) it finishes and the client cann't know why. To know when and why the `OPTIMIZE` query have not performed merge, enable the exception throwing.
```
Comment (blinkov) [inline docs/en/operations/settings/settings.md]:
```suggestion
By default `OPTIMIZE` returns successfully even if it haven't done anything. This setting allows to distinguish this situation and get the reason in exception message.
```

File: docs/en/query_language/misc.md [hunk 3]
```diff
@@ -177,10 +177,16 @@ Changes already made by the mutation are not rolled back.
 OPTIMIZE TABLE [db.]name [ON CLUSTER cluster] [PARTITION partition] [FINAL]
 ```
 
-Asks the table engine to do something for optimization.
-Supported only by `*MergeTree` engines, in which this query initializes a non-scheduled merge of data parts.
-If you specify a `PARTITION`, only the specified partition will be optimized.
-If you specify `FINAL`, optimization will be performed even when all the data is already in one part.
+Supported only by `MergeTree` family of table engines, in which this query initializes a non-scheduled merge of data parts.
```
Comment (blinkov) [inline docs/en/query_language/misc.md]:
```suggestion
This query tries to initialize an unscheduled merge of data parts for tables with a table engine of [MergeTree](../operations/table_engines/mergetree.md) family. Other kinds of table engines are not supported.
```

File: docs/en/query_language/misc.md [hunk 4]
```diff
@@ -177,10 +177,16 @@ Changes already made by the mutation are not rolled back.
...
+
+When used with [MergeTree](../operations/table_engines/mergetree.md) family of table engines `OPTIMIZE` forces data parts merges. ClickHouse tries to find data parts for merging and then merge them. If there is nothing to merge, ClickHouse does nothing.
```
Comment (blinkov) [inline docs/en/query_language/misc.md]: I'd just drop this paragraph

File: docs/en/query_language/misc.md [hunk 5]
```diff
@@ -177,10 +177,16 @@ Changes already made by the mutation are not rolled back.
...
+
+When `OPTIMIZE` used with [ReplicatedMergeTree](../operations/table_engines/replication.md) family of table engines, ClickHouse creates a task for merging and waits for execution on all nodes (if the `replication_alter_partitions_sync` setting is enabled).
```
Comment (blinkov) [inline docs/en/query_language/misc.md]: link to setting?

File: docs/en/query_language/misc.md [hunk 6]
```diff
@@ -177,10 +177,16 @@ Changes already made by the mutation are not rolled back.
...
+
+If `OPTIMIZE` doesn't perform merging for any reason, it doesn't notify the client about it. To enable notification use the [optimize_throw_if_noop](../operations/settings/settings.md#setting-optimize_throw_if_noop) setting.
+
+If you specify a `PARTITION`, only the specified partition is optimized.
```
Comment (blinkov) [inline docs/en/query_language/misc.md]: Group this if's in some bullet point list?

**Review by blinkov — CHANGES_REQUESTED** [review]

File: docs/en/query_language/misc.md (hunk 5 above)
Comment (BayoNet) [reply docs/en/query_language/misc.md]: It's not documented, but it will be.

**Pull request merged**